### PR TITLE
Dp custom order

### PR DESCRIPTION
**Pull request by DenisPopov15**

### CODE REVIEW

File: lib/api/actions.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ class Index extends Base {
   constructor(modelName, options = {}) {
     super(modelName)
 
-    this.sortBy   = options.sortBy   || { createdAt: -1 }
+    this.sortBy   = options.sortBy   || { position: 1, createdAt: -1 }
```
Comment (alexkravets) [inline lib/api/actions.js]: Doesn't look to be the right approach, as requires extra index to be created. I'm thinking about a way to generalize the approach and store position (probably not a good name cause of the context) for all docs based on `createdAt` with an option to update position.

File: lib/db/mongodb/index.js [hunk 2]
```diff
@@ -61,6 +61,9 @@ class Mongodb {
       schema.hasDynamicCollectionName = true
     }
 
+    if (options.customOrder) {
```
Comment (alexkravets) [inline lib/db/mongodb/index.js]: This should not be a schema option, should be removed. If plugin is used by some models, should be defined in a model not on a general level.

File: lib/db/mongodb/plugins/customOrder.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+'use strict'
```
Comment (alexkravets) [inline lib/db/mongodb/plugins/customOrder.js]: The problem with ordering is that we need a general approach of doing this. Creating indexes for many fields is not a good option as there is a limit on index name size. We had an issue with: staff profile: first name, last name, email — index is too long. Also if we generalize the ordering approach, we don't need to create custom indexes.

**Review by alexkravets — CHANGES_REQUESTED** [review]

File: lib/db/mongodb/plugins/customOrder.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
```
Comment (alexkravets) [inline lib/db/mongodb/plugins/customOrder.js]: `orderable`

File: lib/db/mongodb/plugins/orderable.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  const orderableIndex      = { _deleted: 1 }
+  orderableIndex[fieldName] = 1
+
+  schema.add(orderableField)
+
+  const sortBy = {}
+  sortBy[fieldName] = -1
+  schema.pre('save', function (next) {
+    this.constructor.find({}).sort(sortBy).limit(1)
```
Comment (alexkravets) [inline lib/db/mongodb/plugins/orderable.js:16]: Are you sure index is used in this case, do we need to add `_delete: true` or it's added automatically?

**Review by alexkravets — CHANGES_REQUESTED** [review]
Add tests, make sure created index is used.

File: lib/db/mongodb/plugins/orderable.js (hunk 5 above)
Comment (DenisPopov15) [reply lib/db/mongodb/plugins/orderable.js:16]: @alexkravets yes, seems so - here logs from this pre save hook : http://files.slatestudio.com/ql2W (deleted added automatic to query, index is used)

**Comment by DenisPopov15** [comment]
@alexkravets PR updated

**Pull request merged**